Mathlib/CategoryTheory/Distributive/Monoidal.lean

```diff
@@ -0,0 +1,321 @@
+    PreservesColimitsOfShape (Discrete WalkingPair) (tensorLeft X)
+
+/-- A monoidal category with binary coproducts is right distributive
+if the right tensor product functor preserves binary coproducts. -/
+class IsMonoidalRightDistrib (C : Type u) [Category.{v} C]
+    [MonoidalCategory.{v} C] [HasBinaryCoproducts C] : Prop where
+  preservesBinaryCoproducts_tensorRight (X : C) :
+    PreservesColimitsOfShape (Discrete WalkingPair) (tensorRight X)
+
+/-- A monoidal category with binary coproducts is distributive
+if it is both left and right distributive. -/
+class IsMonoidalDistrib (C : Type u) [Category.{v} C]
+    [MonoidalCategory.{v} C] [HasBinaryCoproducts C] extends
+  IsMonoidalLeftDistrib C, IsMonoidalRightDistrib C
+
+variable {C} [Category.{v} C] [MonoidalCategory.{v} C] [HasBinaryCoproducts C]
+
+example [IsMonoidalDistrib C] : IsMonoidalLeftDistrib C := by infer_instance
+
+section IsMonoidalLeftDistrib
+
+instance IsMonoidalLeftDistrib.preserves_binary_coproducts_tensorLeft
+    [IsMonoidalLeftDistrib C] {X : C} :
+    PreservesColimitsOfShape (Discrete WalkingPair) (tensorLeft X) :=
+  IsMonoidalLeftDistrib.preservesBinaryCoproducts_tensorLeft X
+
+instance IsMonoidalLeftDistrib.preservesColimit_pair_tensorLeft
+    [IsMonoidalLeftDistrib C] {X Y Z : C} :
+    PreservesColimit (pair Y Z) (tensorLeft X) :=
+  (IsMonoidalLeftDistrib.preservesBinaryCoproducts_tensorLeft X).preservesColimit
+
+/-- The canonical left distributivity isomorphism -/
+def leftDistrib [IsMonoidalLeftDistrib C] (X Y Z : C) :
+    (X ⊗ Y) ⨿ (X ⊗ Z) ≅ X ⊗ (Y ⨿ Z) :=
+  PreservesColimitPair.iso (tensorLeft X) Y Z
+
+/-- Notation for the forward direction morphism of the canonical left distributivity isomorphism -/
+scoped notation "∂L" => leftDistrib
+
+instance IsMonoidalLeftDistrib.isIso_leftDistrib_hom [IsMonoidalLeftDistrib C] {X Y Z : C} :
+    IsIso (∂L X Y Z).hom :=
+  isIso_hom <| leftDistrib X Y Z
+
+instance IsMonoidalLeftDistrib.of_isIso_coprodComparisonTensorLeft
+    [i : ∀ {X Y Z : C}, IsIso (coprodComparison (tensorLeft X) Y Z)] : IsMonoidalLeftDistrib C where
+  preservesBinaryCoproducts_tensorLeft X := by
+    refine {
+      preservesColimit := by
+        intro K
+        have : PreservesColimit
+          (pair (K.obj { as := WalkingPair.left }) (K.obj { as := WalkingPair.right }))
+          (tensorLeft X) := by
+            apply PreservesColimitPair.of_iso_coprod_comparison (tensorLeft X) _ _
+        apply preservesColimit_of_iso_diagram (tensorLeft X) (diagramIsoPair K).symm
+    }
+
+variable [IsMonoidalLeftDistrib C]
+
+/-- The forward direction of the left distributivity isomorphism is the cogap morphism
+`coprod.desc (_ ◁ coprod.inl) (_ ◁ coprod.inr) : (X ⊗ Y) ⨿ (X ⊗ Z) ⟶ X ⊗ (Y ⨿ Z)`. -/
+@[simp]
+lemma leftDistrib_hom {X Y Z : C} :
+  (∂L X Y Z).hom = coprod.desc (_ ◁ coprod.inl) (_ ◁ coprod.inr) := by rfl
+
+lemma coprod_inl_leftDistrib {X Y Z : C} :
+    coprod.inl ≫ (∂L X Y Z).hom = (X ◁ coprod.inl) := by
```

Comment:
```suggestion
@[reassoc (attr := simp)]
lemma coprod_inl_leftDistrib {X Y Z : C} :
    coprod.inl ≫ (∂L X Y Z).hom = X ◁ coprod.inl := by
```
Same below.

Reply:
I'm not sure about this, since `simp` can prove it. Adding `simp` gives linter error, I believe.